azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/AbstractWebAppMojo.java

```diff
@@ -419,10 +418,105 @@ protected WebAppConfig getWebAppConfig() throws AzureExecutionException {
         final SchemaVersion version = SchemaVersion.fromString(getSchemaVersion());
         final AbstractConfigurationValidator validator = version == SchemaVersion.V2 ?
                 new V2ConfigurationValidator(this) : new V1ConfigurationValidator(this);
-        final AbstractConfigParser parser = version == SchemaVersion.V2 ? new V2ConfigParser(this, validator) : new V2ConfigParser(this, validator);
+        final AbstractConfigParser parser = version == SchemaVersion.V2 ? new V2ConfigParser(this, validator) : new V1ConfigParser(this, validator);
         return parser.parse();
     }
 
+    protected AzureAppService getOrCreateAzureAppServiceClient() throws AzureExecutionException {
```

Comment:
shall we delete getOrCreateAzureClient in class AbstractAzureMojo in future?

Reply:
selectSubscription checkSubscription are very similar with the same class in  AbstractAzureMojo, or shall we move getOrCreateAzureAppServiceClient to AbstractAppServiceMojo and redefine selectSubscription/checkSubscription in  AbstractAzureMojo

Reply:
Yes, there is an issue that our codes for get azure client and select subscription uses models from SDK, and most maven plugin still uses the track one version, while webapp maven plugin uses the model from track2. This is the workaround that create another implement with track2 sdk for webapp only, and we will finally migrate all maven tooling to track2 and replace the duplicate implements.